src/components/Form.js

```diff
@@ -1,35 +1,74 @@
 import React from 'react'
 import styled from 'styled-components/macro'
 import { NavLink, useHistory } from 'react-router-dom'
+import { useForm } from 'react-hook-form'
+import { v4 as uuidv4 } from 'uuid'
 
 export default function Form({ onSubmit }) {
   const history = useHistory()
+  const { register, handleSubmit, errors } = useForm()
+  const onFormSubmit = (newReview) => {
+    onSubmit({ ...newReview, id: uuidv4() })
+    history.push('/')
+  }
   return (
-    <StyledForm onSubmit={handleSubmit}>
+    <StyledForm onSubmit={handleSubmit(onFormSubmit)}>
       <label>
         Title
-        <Input autoFocus name="title" />
+        <Input autoFocus name="title" ref={register({ required: true })} />
       </label>
+      {errors.title && errors.title.type === 'required' && (
+        <StyledErrorMessage>Title is required!</StyledErrorMessage>
+      )}
+
       <label>
-        Rating (0.0 - 10.0)
-        <Input name="rating" />
+        Rating (1-100)
+        <Input
+          name="rating"
+          ref={register({ required: true, pattern: /^[1-9][0-9]?$|^100$/ })}
+        />
       </label>
+      {errors.rating && errors.rating.type === 'required' && (
+        <StyledErrorMessage>Rating is required!</StyledErrorMessage>
+      )}
+      {errors.rating && errors.rating.type === 'pattern' && (
+        <StyledErrorMessage>
+          Please choose a number between 0 and 100
```

Comment:
0 is currently not allowed

```suggestion
          Please choose a number between 1 and 100
```

Reply:
You are correct, it should be 1 in the text as well!